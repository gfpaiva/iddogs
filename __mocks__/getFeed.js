module.exports = {
  category: 'labrador',
  list: [
    'https://images.dog.ceo/breeds/labrador/n02099712_1150.jpg',
    'https://images.dog.ceo/breeds/labrador/n02099712_1200.jpg',
    'https://images.dog.ceo/breeds/labrador/n02099712_1229.jpg',
    'https://images.dog.ceo/breeds/labrador/n02099712_1254.jpg',
    'https://images.dog.ceo/breeds/labrador/n02099712_129.jpg',
  ],
};
