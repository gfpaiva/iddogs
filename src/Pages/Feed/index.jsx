import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import Spinner from 'react-spinkit';

import Grid from '../../Components/Grid';

import { getFeed } from '../../Utils/API';
import CategoryMenu from '../../Components/CategoryMenu';

export default function Feed({ match: { params: { category } } }) {
  const [dogs, setDogs] = useState([]);
  const [loading, seLoading] = useState(true);

  useEffect(() => {
    async function fetchDogs() {
      try {
        seLoading(true);

        const { list } = await getFeed(category);

        setDogs(list);
        seLoading(false);
      } catch {
        setDogs([]);
        seLoading(false);
      }
    }

    fetchDogs();
  }, [category]);

  if (loading) {
    return (
      <div className="text-center">
        <Spinner name="three-bounce" />
      </div>
    );
  }

  if (!dogs || !dogs.length) {
    return (
      <h3 className="text-center">
        N√£o encontramos nenhum cachorro nesta categoria.
        <span role="img" aria-label="Cachorro">üê∂</span>
      </h3>
    );
  }

  return (
    <div>
      <CategoryMenu />
      <Grid
        items={dogs}
      />
    </div>
  );
}

Feed.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      category: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};
