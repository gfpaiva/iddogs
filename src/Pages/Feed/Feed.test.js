import React from 'react';
import wait from 'waait';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';''

import Feed from './index';

import getFeed from '../../../__mocks__/getFeed';

describe('<Feed />', () => {
  it('should mount properly', () => {
		const wrapper = mount(
      <Router>
        <Feed match={{ params: { category: 'husky' } }} />
      </Router>
    );

		expect(wrapper.find('Feed')).toMatchSnapshot();
  });

  it('should mount with data properly', async () => {
    fetch
      .once(JSON.stringify(getFeed));

		const wrapper = mount(
      <Router>
        <Feed match={{ params: { category: 'husky' } }} />
      </Router>
    );

    await wait();
    wrapper.update();

		expect(wrapper.find('CategoryMenu')).toHaveLength(1);
		expect(wrapper.find('Grid')).toHaveLength(1);
  });

  it('should mount with fetch error properly', async () => {
    fetch
      .mockRejectOnce('ERRO')

		const wrapper = mount(
      <Router>
        <Feed match={{ params: { category: 'husky' } }} />
      </Router>
    );

    await wait();
    wrapper.update();

		expect(wrapper.find('Feed')).toMatchSnapshot();
  });
});
