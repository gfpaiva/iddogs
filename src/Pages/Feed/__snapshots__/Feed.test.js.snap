// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feed /> should mount properly 1`] = `
<Feed
  match={
    Object {
      "params": Object {
        "category": "husky",
      },
    }
  }
>
  <div
    className="text-center"
  >
    <Spinner
      fadeIn="full"
      name="three-bounce"
      noFadeIn={false}
      overrideSpinnerClassName=""
    >
      <div
        className="sk-fade-in sk-spinner sk-three-bounce"
      >
        <div
          key="0"
        />
        <div
          key="1"
        />
        <div
          key="2"
        />
      </div>
    </Spinner>
  </div>
</Feed>
`;

exports[`<Feed /> should mount with fetch error properly 1`] = `
<Feed
  match={
    Object {
      "params": Object {
        "category": "husky",
      },
    }
  }
>
  <h3
    className="text-center"
  >
    N√£o encontramos nenhum cachorro nesta categoria.
    <span
      aria-label="Cachorro"
      role="img"
    >
      üê∂
    </span>
  </h3>
</Feed>
`;
